datalad_next/patches/push_to_export_remote.py

```diff
@@ -0,0 +1,145 @@
+    Iterable,
+    Optional,
+    Union,
+)
+from unittest.mock import patch
+
+import datalad.core.distributed.push as push
+from datalad.distribution.dataset import Dataset
+from datalad.runner.exception import CommandError
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.exceptions import CapturedException
+from datalad_next.credman import CredentialManager
+from datalad_next.utils import (
+    get_specialremote_credential_envpatch,
+    get_specialremote_credential_properties,
+    needs_specialremote_credential_envpatch,
+)
+
+
+__docformat__ = "restructuredtext"
+
+
+lgr = logging.getLogger('datalad_next.push_to_export_remote')
+
+
+def _is_export_remote(remote_info: Optional[Dict]) -> bool:
+    """Check if remote_info is valid and has exporttree set to "yes"
+
+    :param remote_info: None or a dictionary the contains git annex special
+    remote info
+    :return: True if exporttree is yes, else False
+    """
+    if remote_info is not None:
+        return remote_info.get("exporttree") == "yes"
+    return False
+
+
+def _is_webdav_remote(repo: AnnexRepo,
+                      target: str,
+                      ) -> bool:
+    """Check if remote "target" exists and has type set to "webdav"
+
+    :param repo: the repository that contains the remote
+    :param target: name of the remote that should be checked
+    :return: True if git annex reports exporttree is yes, else False
+    """
+    remote_info = [
+        record
+        for record in repo.get_special_remotes().values()
+        if record.get("name") == target]
+    if len(remote_info) == 1:
+        return remote_info[0].get("exporttree") == "yes"
+    return False
+
+
+def _transfer_data(repo: AnnexRepo,
+                   ds: Dataset,
+                   target: str,
+                   content: Iterable,
+                   data: str,
+                   force: Optional[str],
+                   jobs: Optional[Union[str, int]],
+                   res_kwargs: Dict,
+                   got_path_arg: bool
+                   ) -> Generator:
+
+    remote_info = ([
+        record
+        for record in repo.get_special_remotes().values()
+        if record.get("name") == target] or [None])[0]
+
+    if _is_export_remote(remote_info):
+        # TODO:
+        #  - check for configuration entries, e.g. what to export
+        #  - check for all kind of things that are checked in push._push_data
+        #  - proper error handling
+
+        lgr.debug("Exporting HEAD to a remote with exporttree == yes")
+
+        if ds.config.getbool('remote.{}'.format(target), 'annex-ignore', False):
+            lgr.debug(
+                "Target '%s' is set to annex-ignore, exclude from data-export.",
+                target)
+            return
+
+        # Check for credentials
+        sr_params = {
+            "type": remote_info.get("type"),
+            "url": remote_info.get("url")
+        }
+        credential_name, credentials = None, None
```

Comment:
Technically, we have no use for `credential_name`

Reply:
Unless we also do the update-on-success pattern

Reply:
will remove it. I think updates should be left to create-sibling-webdav